keras/callbacks.py

```diff
@@ -571,7 +571,11 @@ def __init__(self, schedule, verbose=0):
     def on_epoch_begin(self, epoch, logs=None):
         if not hasattr(self.model.optimizer, 'lr'):
             raise ValueError('Optimizer must have a "lr" attribute.')
-        lr = self.schedule(epoch)
+        lr = float(K.get_value(self.model.optimizer.lr))
+        try:  # new API
+            lr = self.schedule(epoch, lr)
```

Comment:
Better to use keyword arguments (same below). API change looks ok to me.

Reply:
Done, assuming I understood you correctly. This is a breaking change, since we now require user-passed `schedule()` to have argument named `epoch`, which we didn't before.

Reply:
That's a good point, since we didn't require it be named `epoch` before, we should probably make the first argument positional and only make the second (new one) a keyword arg.

Reply:
I can relax this only in the old API. New one can require both to be named.